openedx/features/enterprise_support/api.py

```diff
@@ -572,7 +572,12 @@ def consent_needed_for_course(request, user, course_id, enrollment_exists=False)
         )
     else:
         client = ConsentApiClient(user=request.user)
-        current_enterprise_uuid = enterprise_customer_uuid_for_request(request)
+
+        current_enterprise_uuid = None
+        if enterprise_learner_details:
+            learner_active_enterprise = get_learner_active_enterprise(enterprise_learner_details)
+            current_enterprise_uuid = learner_active_enterprise['uuid']
```

Comment:
**Note to Reviewers:** We already have learner's active enterprise detail in `enterprise_learner_details` so we don't need to call `enterprise_customer_uuid_for_request`.